paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/FormViewModel.kt

```diff
@@ -28,35 +33,62 @@ import javax.inject.Singleton
  * @param: layout - this contains the visual layout of the fields on the screen used by [Form]
  * to display the UI fields on screen.  It also informs us of the backing fields to be created.
  */
-@Singleton
 internal class FormViewModel @Inject internal constructor(
     layout: LayoutSpec,
     config: FormFragmentArguments,
-    private val resourceRepository: ResourceRepository
+    resourceRepository: ResourceRepository
 ) : ViewModel() {
     internal class Factory(
-        private val resources: Resources,
-        private val layout: LayoutSpec,
-        private val formArguments: FormFragmentArguments
-    ) : ViewModelProvider.Factory {
+        val config: FormFragmentArguments,
+        val resource: Resources,
+        var layout: LayoutSpec
+    ) : ViewModelProvider.Factory, Injectable<Factory.FallbackInitializeParam> {
+        internal data class FallbackInitializeParam(
+            val resource: Resources
+        )
+
+        @Inject
+        lateinit var subComponentBuilderProvider: Provider<FormViewModelSubcomponent.Builder>
+
         @Suppress("UNCHECKED_CAST")
         override fun <T : ViewModel?> create(modelClass: Class<T>): T {
 
-            // This is where we will call Dagger:
-            return DaggerFormViewModelComponent.builder()
-                .resources(resources)
+            val logger = Logger.getInstance(BuildConfig.DEBUG)
+            WeakMapInjectorRegistry.retrieve(config.injectorKey)?.let {
+                logger.info(
+                    "Injector available, " +
+                        "injecting dependencies into FormViewModel.Factory"
+                )
+                it.inject(this)
+            } ?: run {
+                logger.info(
+                    "Injector unavailable, " +
+                        "initializing dependencies of FormViewModel.Factory"
+                )
+                fallbackInitialize(
+                    FallbackInitializeParam(
+                        resource
+                    )
+                )
+            }
+            return subComponentBuilderProvider.get()
+                .formFragmentArguments(config)
                 .layout(layout)
-                .formFragmentArguments(formArguments)
+                .build().viewModel as T
+        }
+
+        override fun fallbackInitialize(arg: FallbackInitializeParam) {
+            DaggerFormViewModelComponent.builder()
+                .resources(arg.resource)
                 .build()
-                .viewModel as T
+                .inject(this)
         }
     }
 
     private val transformSpecToElement = TransformSpecToElement(resourceRepository, config)
```

Comment:
(not required in this PR) - now that `FormViewModel` is fully injected, this field could be injected too. Since both fields are already provided, you just need to annotate `TransformSpecToElement`'s constructor with `@Inject` and declare it in `FormViewModel`'s constructor